addons/payment_stripe/static/src/js/express_checkout_form.js

```diff
@@ -0,0 +1,267 @@
+                        } else {
+                            // The payment has succeeded.
+                            window.location = '/payment/status';
+                        }
+                    } else {
+                        // The payment has succeeded.
+                        window.location = '/payment/status';
+                    }
+                }
+            });
+
+            if(isShippingInformationRequired) {
+                this.stripeExpressCheckoutForm.on('shippingaddresschange', async (ev) => {
+                    // Perform server-side request to fetch shipping options
+                    const availableCarriers = await this._rpc({
+                        route: '/shop/express/shipping_address_change',
+                        params: {
+                            shipping_address: {
+                                zip: ev.shippingAddress.postalCode,
+                                city: ev.shippingAddress.city,
+                                country: ev.shippingAddress.country,
+                                state: ev.shippingAddress.region,
+                            },
+                        },
+                    });
+                    if (availableCarriers.length === 0) {
+                        ev.updateWith({status: 'invalid_shipping_address'});
+                    } else {
+                        ev.updateWith({
+                            status: 'success',
+                            shippingOptions: availableCarriers.map(carrier => ({
+                                id: String(carrier.id),
+                                label: carrier.name,
+                                detail: carrier.description ? carrier.description:"",
+                                amount: carrier.minorAmount,
+                            })),
+                            total: {
+                                label: this.txContext.label,
+                                amount: this.txContext.minorAmount + availableCarriers[0].minorAmount,
+                                pending: false,
+                            },
+                            displayItems: [
+                                {
+                                    label: _t('Your order'),
+                                    amount: this.txContext.minorAmount,
+                                },
+                                {
+                                    label: _t('Delivery'),
+                                    amount: availableCarriers[0].minorAmount,
+                                    pending: false,
+                                },
+                            ],
+                        });
+                    }
+                });
+
+                this.stripeExpressCheckoutForm.on('shippingoptionchange', async (ev) => {
+                    ev.updateWith({
+                        //TODO Check with server here ?
```

Comment:
TODO left here

Reply:
I know, I want to check with @AntoineVDV so I purposely left it here ^^ It will be removed before merging. Thanks for noticing it :)